modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -443,8 +443,19 @@ private void updateLeaseBatchInternal() {
                             ? lease.getLeaseholder()
                             : lease.proposedCandidate();
 
-                    ClusterNode candidate = nextLeaseHolder(assignments, grpId, proposedLeaseholder);
+                    ClusterNode candidate = nextLeaseHolder(stableAssignments, grpId, proposedLeaseholder);
 
+                    // If there wasn't a candidate among stable assignments set then make attempt to select a candidate among pending set
+                    if (candidate == null) {
```

Comment:
This part has to be packed to the nextLeaseHolder method.

Reply:
Done